central/splunk/compliance_test.go

```diff
@@ -0,0 +1,188 @@
+							Namespace: "dep-ns1",
+						},
+					},
+					Nodes: map[string]*storage.Node{
+						"node1": {
+							Id:   "node1",
+							Name: "node1",
+						},
+					},
+				},
+				ClusterResults: &storage.ComplianceRunResults_EntityResults{
+					ControlResults: map[string]*storage.ComplianceResultValue{
+						"HIPAA_164:310_a_1": {
+							Evidence: []*storage.ComplianceResultValue_Evidence{{
+								State:   storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+								Message: "Cluster has an image scanner in use",
+							}},
+							OverallState: storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+						},
+					},
+				},
+				DeploymentResults: map[string]*storage.ComplianceRunResults_EntityResults{
+					"deployment1": {
+						ControlResults: map[string]*storage.ComplianceResultValue{
+							"CIS_Docker_v1_2_0:5_6": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+									Message: "Container has no ssh process running",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+							},
+						},
+					},
+				},
+				NodeResults: map[string]*storage.ComplianceRunResults_EntityResults{
+					"node1": {
+						ControlResults: map[string]*storage.ComplianceResultValue{
+							"CIS_Docker_v1_2_0:1_1_2": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SKIP,
+									Message: "Node does not use Docker container runtime",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SKIP,
+							},
+						},
+					},
+				},
+				MachineConfigResults: map[string]*storage.ComplianceRunResults_EntityResults{
+					"ocp4-cis-node-master": {
+						ControlResults: map[string]*storage.ComplianceResultValue{
+							"ocp4-cis-node:file-owner-worker-kubeconfig": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+									Message: "Pass for ocp4-cis-node-master-file-owner-worker-kubeconfig.",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestSplunkComplianceAPI(t *testing.T) {
+	suite.Run(t, &splunkComplianceAPITestSuite{})
+}
+
+type splunkComplianceAPITestSuite struct {
+	suite.Suite
+
+	hasReadCtx  context.Context
+	hasWriteCtx context.Context
+
+	mockCtrl *gomock.Controller
+	mockDS   *mocks.MockDataStore
+}
+
+func (s *splunkComplianceAPITestSuite) SetupTest() {
+	s.hasReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Compliance)))
+	s.mockCtrl = gomock.NewController(s.T())
+	s.mockDS = mocks.NewMockDataStore(s.mockCtrl)
+}
+
+func (s *splunkComplianceAPITestSuite) TearDownTest() {
+	s.mockCtrl.Finish()
+}
```

Comment:
Just checking to make sure, do `s.mockOS` and `s.hasReadCtx` not need to be reset between tests?

Reply:
Basically, I'm setting all structures up just once and save the response of the http call in a variable.
The singled out tests are all running on the cached data, so there should be no reset needed.
To be brutal, one could say these tests are just glorified string checking on a single, cached result string :)

Additionally, the tests aren't interested in the results of these objects, so there should be no need to rest them between tests.

Reply:
Well spotted, though! That would definitely be a concern if we were interested in the exact call behaviour 👍